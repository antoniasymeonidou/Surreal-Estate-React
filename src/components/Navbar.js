import React, { useState } from "react";
import { Link } from "react-router-dom";
import FacebookLogin from "react-facebook-login";
import PropTypes from "prop-types";
import { FaFortAwesome } from "react-icons/fa";
import "../styles/Navbar.css";

export default function Navbar({ onLogin, userID, onLogout, name }) {
  const [open, setOpen] = useState(false);

  const handleNav = (e) => {
    e.preventDefault();
    setOpen(!open);
  };

  return (
    <nav>
      <Link to="/" className="logo">
        <FaFortAwesome className="logo-img" data-testid="logo-id" />
        <p className="logo-title">Surreal Estate</p>
      </Link>

      <ul className={`collapsed ${open ? "open" : ""}`}>
        <li>
          <Link
            to="/properties"
            className="item"
            data-testid="view-properties-id"
          >
            View Properties
          </Link>
        </li>
        <li>
          <Link
            to="/add-property"
            className="item"
            data-testid="add-property-id"
          >
            Add Property
          </Link>
        </li>
        <li>
          <Link to="/favourites" className="item" data-testid="favourites-id">
            Favourites
          </Link>
        </li>
        <li className="facebook-container">
          {userID ? (
            <button
              onClick={onLogout}
              className="my-facebook-button-class"
              type="submit"
            >
              Welcome,
              <br />
              {name}
              <br />
              Sign Out
            </button>
          ) : (
            <FacebookLogin
              appId="651558712097637"
              fields="name,email,picture"
              callback={onLogin}
              cssClass="my-facebook-button-class"
            />
          )}
        </li>
      </ul>
      <button type="button" className="burger" onClick={(e) => handleNav(e)}>
        <div className="line1" />
        <div className="line2" />
        <div className="line3" />
      </button>
    </nav>
  );
}

Navbar.propTypes = {
  onLogin: PropTypes.func.isRequired,
  userID: PropTypes.string.isRequired,
  onLogout: PropTypes.func.isRequired,
  name: PropTypes.string,
};