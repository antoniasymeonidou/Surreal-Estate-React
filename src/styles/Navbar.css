nav {
    grid-column: 1 / 4;
    grid-row: 1 / 2;
    display: flex;
    justify-content: space-around;
    align-items: center;
  }
  
  .logo {
    display: flex;
    align-items: baseline;
    white-space: nowrap;
    text-decoration: none;
    color: black;
    font-size: 25px;
  }
  
  .logo-img {
    font-size: 40px;
    color: rgb(93, 9, 104);
    opacity: 1;
  }
  
  .logo-title {
    margin-left: 3px;
  }
  
  .logo-img:hover {
    opacity: 0.5;
  }
  
  .collapsed {
    display: flex;
    width: 70%;
    justify-content: space-around;
    align-items: center;
    list-style: none;
  }
  
  .item {
    text-decoration: none;
    letter-spacing: 2px;
    color: black;
  }
  
  .item:hover {
    color: #928583;
  }
  
  .my-facebook-button-class {
    background-color: #3b5998;
    color: white;
    display: flex;
    padding: 5px;
    width: 100%;
    height: 60px;
    justify-content: space-evenly;
    align-items: center;
  }
  
  .my-facebook-button-class:hover {
    background-color: #abb6cf;
  }
  
  .burger {
    display: none;
    cursor: pointer;
    background-color: transparent;
    border: none;
  }
  
  .burger div {
    width: 25px;
    height: 3px;
    margin: 5px;
    background-color: rgb(161, 62, 179);
  }
  
  @media only screen and (max-width: 400px) {
    nav {
      background-color: rgb(245, 234, 239);
    }
  
    .logo {
      font-size: 18px;
      margin-left: 10px;
    }
  
    .logo-img {
      font-size: 30px;
    }
  
    .item {
      font-size: 15px;
    }
  
    ul.collapsed {
      display: flex;
      flex-direction: column;
      justify-content: space-around;
  
      overflow: hidden;
      max-height: 0;
      -o-transition-duration: 0.4s;
      transition-duration: 0.4s;
      -o-transition-timing-function: cubic-bezier(0, 1, 0.5, 1);
      transition-timing-function: cubic-bezier(0, 1, 0.5, 1);
    }
    ul.collapsed.open {
      overflow: hidden;
      max-height: 92vh; /* approximate max height */
      -o-transition-duration: 0.4s;
      transition-duration: 0.4s;
      -o-transition-timing-function: ease-in;
      transition-timing-function: ease-in;
    }
  
    li {
      padding-top: 20px;
      width: 100%;
    }
  
    .burger {
      display: block;
    }
  
    .my-facebook-button-class {
      height: auto;
    }
  }