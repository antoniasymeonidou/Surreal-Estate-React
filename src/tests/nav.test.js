import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Navbar from "../components/Navbar";

describe("Navbar", () => {
  const mockFacebookLogin = jest.fn();
  const mockLoginData = {
    userID: "userID",
    picture: "picture",
    email: "email",
    name: "name",
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Navbar
          {...mockLoginData}
          onLogin={mockFacebookLogin}
          onLogout={mockFacebookLogin}
        />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct specs", () => {
    const { getByTestId } = render(
      <MemoryRouter>
        <Navbar
          {...mockLoginData}
          onLogin={mockFacebookLogin}
          onLogout={mockFacebookLogin}
        />
      </MemoryRouter>
    );
    const logo = getByTestId("logo-id");
    expect(logo).toBeInTheDocument();

    const viewProperties = getByTestId("view-properties-id");
    expect(viewProperties).toHaveTextContent("View Properties");

    const addProperty = getByTestId("add-property-id");
    expect(addProperty).toHaveTextContent("Add Property");

    const favourites = getByTestId("favourites-id");
    expect(favourites).toHaveTextContent("Favourites");
  });
});