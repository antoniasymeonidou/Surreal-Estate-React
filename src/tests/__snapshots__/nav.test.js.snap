// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
<DocumentFragment>
  <nav>
    <a
      class="logo"
      href="/"
    >
      <svg
        class="logo-img"
        data-testid="logo-id"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M489.2 287.9h-27.4c-2.6 0-4.6 2-4.6 4.6v32h-36.6V146.2c0-2.6-2-4.6-4.6-4.6h-27.4c-2.6 0-4.6 2-4.6 4.6v32h-36.6v-32c0-2.6-2-4.6-4.6-4.6h-27.4c-2.6 0-4.6 2-4.6 4.6v32h-36.6v-32c0-6-8-4.6-11.7-4.6v-38c8.3-2 17.1-3.4 25.7-3.4 10.9 0 20.9 4.3 31.4 4.3 4.6 0 27.7-1.1 27.7-8v-60c0-2.6-2-4.6-4.6-4.6-5.1 0-15.1 4.3-24 4.3-9.7 0-20.9-4.3-32.6-4.3-8 0-16 1.1-23.7 2.9v-4.9c5.4-2.6 9.1-8.3 9.1-14.3 0-20.7-31.4-20.8-31.4 0 0 6 3.7 11.7 9.1 14.3v111.7c-3.7 0-11.7-1.4-11.7 4.6v32h-36.6v-32c0-2.6-2-4.6-4.6-4.6h-27.4c-2.6 0-4.6 2-4.6 4.6v32H128v-32c0-2.6-2-4.6-4.6-4.6H96c-2.6 0-4.6 2-4.6 4.6v178.3H54.8v-32c0-2.6-2-4.6-4.6-4.6H22.8c-2.6 0-4.6 2-4.6 4.6V512h182.9v-96c0-72.6 109.7-72.6 109.7 0v96h182.9V292.5c.1-2.6-1.9-4.6-4.5-4.6zm-288.1-4.5c0 2.6-2 4.6-4.6 4.6h-27.4c-2.6 0-4.6-2-4.6-4.6v-64c0-2.6 2-4.6 4.6-4.6h27.4c2.6 0 4.6 2 4.6 4.6v64zm146.4 0c0 2.6-2 4.6-4.6 4.6h-27.4c-2.6 0-4.6-2-4.6-4.6v-64c0-2.6 2-4.6 4.6-4.6h27.4c2.6 0 4.6 2 4.6 4.6v64z"
        />
      </svg>
      <p
        class="logo-title"
      >
        Surreal Estate
      </p>
    </a>
    <ul
      class="collapsed "
    >
      <li>
        <a
          class="item"
          data-testid="view-properties-id"
          href="/properties"
        >
          View Properties
        </a>
      </li>
      <li>
        <a
          class="item"
          data-testid="add-property-id"
          href="/add-property"
        >
          Add Property
        </a>
      </li>
      <li>
        <a
          class="item"
          data-testid="favourites-id"
          href="/favourites"
        >
          Favourites
        </a>
      </li>
      <li
        class="facebook-container"
      >
        <button
          class="my-facebook-button-class"
          type="submit"
        >
          Welcome,
          <br />
          name
          <br />
          Sign Out
        </button>
      </li>
    </ul>
    <button
      class="burger"
      type="button"
    >
      <div
        class="line1"
      />
      <div
        class="line2"
      />
      <div
        class="line3"
      />
    </button>
  </nav>
</DocumentFragment>
`;
